stack/rest/src/main/java/org/apache/usergrid/rest/management/ManagementResource.java

```diff
@@ -441,6 +451,181 @@ public Viewable handleAuthorizeForm( @Context UriInfo ui, @FormParam( "response_
+        String email    = userNode.get( "email" ).getTextValue();
+
+        // set dummy password to random string that nobody can guess, in SSO setup
+        // admin users should never be able to login directly to this Usergrid system
+        String dummyPassword = RandomStringUtils.randomAlphanumeric( 40 );
+
+        // if user does not exist locally then we need to fix that
+
+        final UUID userId;
+        final OrganizationInfo organizationInfo = management.getOrganizationByName(username);
+
+        if ( organizationInfo == null ) {
+
+            // create local user and personal organization, activate user.
+
+            OrganizationOwnerInfo ownerOrgInfo = management.createOwnerAndOrganization(
+                    username, username, name, email, dummyPassword, true, true );
+            userId = ownerOrgInfo.getOwner().getUuid();
+
+            management.activateOrganization( ownerOrgInfo.getOrganization() );
+
+            applicationCreator.createSampleFor( ownerOrgInfo.getOrganization() );
+
+        } else {
+            userId = management.getAdminUserByUsername( username ).getUuid();
+        }
+
+        // store the external access_token as if it were one of our own
+        management.importTokenForAdminUser( userId, extAccessToken, ttl );
+
+        // success! return JSON object with access_token field
+        AccessInfo accessInfo = new AccessInfo()
+                .withExpiresIn( tokens.getMaxTokenAgeInSeconds(extAccessToken ) )
+                .withAccessToken( extAccessToken );
+
+        return Response.status( SC_OK ).type( jsonMediaType( callback ) ).entity( accessInfo ).build();
+    }
+
+
+    /**
+     * Look up Admin User via UG Central's /management/me endpoint.
+     *
+     * @param extAccessToken Access token issued by UG Central of Admin User
+     * @return JsonNode representation of AccessInfo object for Admin User
+     * @throws EntityNotFoundException if access_token is not valid.
+     */
+    private JsonNode getMeFromUgCentral( String extAccessToken )  throws EntityNotFoundException {
+
+        // create URL of central Usergrid's /management/me endpoint
+
+        String externalUrl = properties.getProperty( USERGRID_CENTRAL_URL ).trim();
+        // be lenient about trailing slash
+        externalUrl = !externalUrl.endsWith( "/" ) ? externalUrl + "/" : externalUrl;
+        String me = externalUrl + "management/me?access_token=" + extAccessToken;
+
+        // use our favorite HTTP client to GET /management/me
+
+        ClientConfig clientConfig = new DefaultClientConfig();
+        clientConfig.getFeatures().put( JSONConfiguration.FEATURE_POJO_MAPPING, Boolean.TRUE);
+        Client client = Client.create(clientConfig);
+        final JsonNode accessInfoNode;
+        try {
+            accessInfoNode = client.resource( me )
+                    .type( MediaType.APPLICATION_JSON_TYPE)
+                    .get(JsonNode.class);
+
+        } catch ( Exception e ) {
```

Comment:
can we expose more specific error information, and throw different exceptions, here based on what comes back from the request?